Test fix php interop test composer install git clone problem

stanley-cheung opened the pull request:
- Fixes #4212 
- Previous fix #3818 and #3859. The path to the `auth.json` file in the host jenkins instance (currently grpc-canary-interop1) has been changed from `/var/local/.composer/auth.json` to `/home/jenkins/.composer/auth.json`.
- With this file properly mounted into the docker image, the `git clone stanley-cheung/Protobuf-PHP` command spawned by the `composer install` command shouldn't fail anymore

grpc-jenkins commented:
Can one of the admins verify this patch?

grpc-jenkins commented:
Can one of the admins verify this patch?

stanley-cheung commented:
I have run the jenkins interop pull requests build 5 times in a row and all come back green.

https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/2040/
https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/2041/
https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/2043/
https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/2044/
https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/2045/
https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/2046/

jtattermusch commented on tools/run_tests/run_interop_tests.py:
```diff
@@ -536,9 +536,9 @@ def build_interop_image_jobspec(language, tag=None):
   # error when running the PHP `composer install` command
   # TODO(stanleycheung): find a more elegant way to do this
-  if language.safename == 'php' and os.path.exists('/var/local/.composer/auth.json'):
+  if language.safename == 'php' and os.path.exists('/home/jenkins/.composer/auth.json'):
```
could we at least use current user's home dir instead of hardcoding /home/jenkins? That way the workaround can work for folks running the interop tests with docker locally.

jtattermusch commented:
If our interop tests not being flaky rely on auth.json being available on the interop test worker, the instructions for setting up the worker need to be updated accordingly (go/grpc-jenkins-setup)

stanley-cheung commented:
* Replace with the ```$HOME``` environment variable
 * Updated the jenkins maintenance log about having to restart the grpc-canary-interop1 instance
 * Updated go/grpc-jenkins-setup doc about having to do this if we set up a new interop jenkins instance

jtattermusch commented on tools/run_tests/run_interop_tests.py:
```diff
@@ -535,10 +535,10 @@ def build_interop_image_jobspec(language, tag=None):
   # error when running the PHP `composer install` command
-  # TODO(stanleycheung): find a more elegant way to do this
-  if language.safename == 'php' and os.path.exists('/var/local/.composer/auth.json'):
+  host_file = '%s/.composer/auth.json' % os.environ['HOME']
```
I think python has a function to tell you the home dir of current user, which would be more portable, but fair enough.
